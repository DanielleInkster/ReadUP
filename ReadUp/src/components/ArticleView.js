import React from 'react';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

Icon.loadFont();

const ArticleView = ({article, deleteEntry}) => {
  return (
    <TouchableOpacity style={styles.articleItem}>
      <View style={styles.articleItemView}>
        <Text style={styles.text}>
          {article.title.substring(0, 30).trim()}...
        </Text>
        <Icon
          name="remove"
          style={styles.icon}
          onPress={() => deleteEntry(article)}
          testID="removeArticle"
        />
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  articleItem: {
    padding: 20,
    backgroundColor: '#bbe1fa',
    borderBottomWidth: 1,
    borderColor: '#eee',
    marginVertical: '1%',
  },
  articleItemView: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  text: {
    color: '#1b262c',
    fontSize: 18,
  },
  icon: {
    fontSize: 25,
    color: 'firebrick',
  },
});

export default ArticleView;
