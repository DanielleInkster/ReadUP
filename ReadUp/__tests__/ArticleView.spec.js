import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import ArticleView from '../src/components/ArticleView';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<ArticleView />).toJSON();
  expect(tree).toMatchSnapshot();
});
