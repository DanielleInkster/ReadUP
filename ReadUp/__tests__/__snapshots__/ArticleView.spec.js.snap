// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#bbe1fa",
      "borderBottomWidth": 1,
      "borderColor": "#eee",
      "borderRadius": 10,
      "marginHorizontal": "2%",
      "marginVertical": "3%",
      "opacity": 1,
      "padding": 20,
    }
  }
  testID="listItem"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1b262c",
          "fontFamily": "sans-serif-medium",
          "fontSize": 18,
        }
      }
    >
      hello
      ...
    </Text>
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "firebrick",
            "fontSize": 25,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="removeArticle"
    >
      ÔÄç
    </Text>
  </View>
</View>
`;
